### DISCUSSION TITLE
How can we get the example of laser melt pool in MOOSE homepage?

### DISCUSSION THREAD

**Discussion by Salma-Mao** [idaholab/moose#20341] [Q&A Getting Started] [answered]
Dear everyone,

I am new to MOOSE. I am trying the simulate the molten pool of selective laser melting. And I find a video about laser melt pool in MOOSE homepage. Anyone can share me the example.

thanks,
MS

**Comment by GiudGiud** [comment]
@cticenhour this was generated with Baldr I think

**Comment by jiangwen84** [comment]
I do not believe we have a laser melt pool example in the MOOSE repo. We developed ALE and Level-set based melt pool in a MOOSE-based application, called Malamute (was named as Baldr and Vahalla before).  @cticenhour do you know if it is possible for external users to get access to it?

**Comment by cticenhour** [comment, marked as answer]
EDIT: MALAMUTE is now open source (no INL HPC account required). Please see https://github.com/idaholab/malamute

The process starts with obtaining an INL HPC account. MALAMUTE isn't under the new access control procedures, so I would have to double-check to see what the process would be for us. I can report back on this. 

One note on the Baldr capability - we did have to update it from a much earlier version of MOOSE, so aspects of those models aren't quite working again yet. @lindsayad and I have been working on it as we've had time. I can also report back once we've got something more stable there.

**Reply by Salma-Mao** [reply]
Thank you very much！Look forward to the work in the future.

**Comment by Eng-Mauvec** [comment]
Hi, I am also interested on utilising MOOSE to simulate the melt pool during SLM. I have been trying to utilise the Heat Conduction module to mimic the latter but cannot find the way to apply the heat input from the laser as a boundary condition. Hence, may I also ask further instructions/information about getting access to MALAMUTE and the melt-pool example previously shown in the MOOSE homepage?

Many thanks,

M

**Reply by jiangwen84** [reply]
We will let you know when we have more information on getting the access to MALAMUTE.

Let me try to help you with your first question regarding how to apply the heat input as a boundary condition. Could you provide a bit more information on what you are trying to do?

**Reply by Salma-Mao** [reply]
There is a example https://mooseframework.inl.gov/source/materials/FunctionPathEllipsoidHeatSource.html. You can change the example to Gaussian heat source  of SLM. If you have new idea, we can exchange the opinion.

**Reply by Eng-Mauvec** [reply]
@jiangwen84 thank your for your response. Yes, I would like to apply a single BCs kernel that contains multiple heat transfer phenomena contributions e.g. convective + radiation, and/or the option to add more such as beam power. However, I only found independent BCs kernels for the radiation and convection.

May I also ask if there's a way to couple Thermo-Calc databases into MOOSE in order to retrieve temperature-dependent material properties, thermodynamic and mobility data  into the Heat Conduction and/or Phase-field modules?

Second question, is there a way to also account for phase transformations within the Heat Conduction module?

@shuine-09 thank you for your suggestion, I was not aware about this Material kernel. I just recently started to explore the in-built heat conduction kernels in MOOSE before aiming to create alternative kernels. An initial idea was to perform the coordinate transformation approach published here: https://www.sciencedirect.com/science/article/pii/S1359645419307979.
